config/base/xnamespace/composition.yaml

```diff
@@ -0,0 +1,66 @@
+---
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: eaas-namespace
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+spec:
+  writeConnectionSecretsToNamespace: crossplane-connections
+  compositeTypeRef:
+    apiVersion: eaas.konflux-ci.dev/v1alpha1
+    kind: XNamespace
+  mode: Pipeline
+  pipeline:
+    - step: create-namespace
+      functionRef:
+        name: function-go-templating
+      input:
+        apiVersion: gotemplating.fn.crossplane.io/v1beta1
+        kind: GoTemplate
+        source: FileSystem
+        fileSystem:
+          dirPath: /templates/xnamespace-ns.yaml
```

Comment:
not sure I follow how this works - how is this file available to the Crossplane controller?

Reply:
They're [mounted](https://github.com/konflux-ci/crossplane-control-plane/pull/10/files#diff-c40a1fc65c59338d7ff9c6577eedbbe20a4e8790af4673170eaa8d84bb129fc3R16-R23) to a volume from a ConfigMap which is generated [here](https://github.com/konflux-ci/crossplane-control-plane/pull/10/files#diff-54495cfee25ec032b3d8ac40e77d0553adaf7513b1bdd4e523f290140594f0fbR7-R13).

Reply:
where is the documentation for doing that> I couldn't find it. Also the filenames and the map key names being different is a bit confusing

Reply:
It's not well documented by `go-templating-function` but there is an [example](https://github.com/crossplane-contrib/function-go-templating/tree/main/example/filesystem).

To avoid confusion with the key names, I modified the configmapgenerator to create an xnamespace specific ConfigMap which now gets mounted to `/templates/xnamespace`.